apps/src/aichat/views/CopyButton.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import {useSelector} from 'react-redux';
+
+import Button from '@cdo/apps/templates/Button';
+import copyToClipboard from '@cdo/apps/util/copyToClipboard';
+import {AichatState} from '@cdo/apps/aichat/redux/aichatRedux';
+
+const CopyButton: React.FunctionComponent = () => {
+  const storedMessages = useSelector(
+    (state: {aichat: AichatState}) => state.aichat.chatMessages
+  );
+
+  const handleCopy = () => {
+    const textToCopy = storedMessages
+      .map(
+        message =>
+          `[${message.timestamp || 'XXXX-XX-XX XX:XX'} - ${message.role}] ${
+            message.chatMessageText
+          }`
+      )
+      .join('\n');
+    copyToClipboard(
+      textToCopy,
+      () => alert('Text copied to clipboard'),
+      () => {
+        console.error('Error in copying text');
+      }
+    );
+  };
+
+  return (
+    <Button
+      color={Button.ButtonColor.white}
+      key="copy"
+      onClick={() => handleCopy()}
+      size={Button.ButtonSize.small}
+      text="Copy Conversation History"
```

Comment:
we could maybe add  `icon="clipboard"` but this is intended to be so short-lived totally up to you!

Reply:
I had that initially, but Michael suggested removing it. Now that I see that both "Version History" and "Show Code" have icons, I'll re-add the icon for consistency!